pkg/exchange/bitget/exchange.go

```diff
@@ -182,9 +184,105 @@ func (e *Exchange) QueryAccountBalances(ctx context.Context) (types.BalanceMap,
 	return bals, nil
 }
 
+// SubmitOrder submits an order.
+//
+// Remark:
+// 1. We support only GTC for time-in-force, because the response from queryOrder does not include time-in-force information.
+// 2. For market buy orders, the size unit is quote currency, whereas the unit for order.Quantity is in base currency.
+// Therefore, we need to calculate the equivalent quote currency amount based on the ticker data.
 func (e *Exchange) SubmitOrder(ctx context.Context, order types.SubmitOrder) (createdOrder *types.Order, err error) {
-	// TODO implement me
-	panic("implement me")
+	if len(order.Market.Symbol) == 0 {
+		return nil, fmt.Errorf("order.Market.Symbol is required: %+v", order)
+	}
+
+	req := e.v2Client.NewPlaceOrderRequest()
+	req.Symbol(order.Market.Symbol)
+
+	// set order type
+	orderType, err := toLocalOrderType(order.Type)
+	if err != nil {
+		return nil, err
+	}
+	req.OrderType(orderType)
+
+	// set side
+	side, err := toLocalSide(order.Side)
+	if err != nil {
+		return nil, err
+	}
+	req.Side(side)
+
+	// set quantity
+	orderQty := order.Quantity
+	// if the order is market buy, the quantity is quote coin, instead of base coin. so we need to convert it.
+	if order.Type == types.OrderTypeMarket && order.Side == types.SideTypeBuy {
+		ticker, err := e.QueryTicker(ctx, order.Market.Symbol)
+		if err != nil {
+			return nil, err
+		}
+		orderQty = order.Quantity.Mul(ticker.Buy)
+	}
+	req.Size(order.Market.FormatQuantity(orderQty))
+
+	// set price
+	switch order.Type {
+	case types.OrderTypeLimit:
+		req.Price(order.Market.FormatPrice(order.Price))
+	}
+
+	// we support only GTC
+	if order.TimeInForce != types.TimeInForceGTC {
+		return nil, fmt.Errorf("time-in-force %s not supported", order.TimeInForce)
+	}
+	req.Force(v2.OrderForceGTC)
+
+	// set client order id
+	if len(order.ClientOrderID) > maxOrderIdLen {
+		return nil, fmt.Errorf("unexpected length of order id, got: %d", len(order.ClientOrderID))
+	}
+	if len(order.ClientOrderID) > 0 {
+		req.ClientOrderId(order.ClientOrderID)
+	}
+
+	if err := submitOrdersRateLimiter.Wait(ctx); err != nil {
+		return nil, fmt.Errorf("place order rate limiter wait error: %w", err)
+	}
+	res, err := req.Do(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to place order, order: %#v, err: %w", order, err)
+	}
+
+	if len(res.OrderId) == 0 || (len(order.ClientOrderID) != 0 && res.ClientOrderId != order.ClientOrderID) {
+		return nil, fmt.Errorf("unexpected order id, resp: %#v, order: %#v", res, order)
+	}
+
+	orderId := res.OrderId
+	ordersResp, err := e.v2Client.NewGetUnfilledOrdersRequest().OrderId(orderId).Do(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query open order by order id: %s, err: %w", orderId, err)
+	}
+
+	switch len(ordersResp) {
```

Comment:
can be simply if len()... {} statement?

Reply:
Yes, it can. But i think switch is more readable than if-else. 

Because we need to check the order has already been filled by NewGetHistoryOrdersRequest